nlp/build-model-worker-on-aws-batch/src/main/kotlin/WorkerOnAwsBatchProvider.kt

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (C) 2017/2020 e-voyageurs technologies
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ai.tock.nlp.build.aws
+
+import ai.tock.nlp.build.ondemand.WorkerOnDemand
+import ai.tock.nlp.build.ondemand.WorkerOnDemandProvider
+import ai.tock.nlp.build.ondemand.WorkerOnDemandType
+import ai.tock.nlp.build.ondemand.WorkerProperties
+import ai.tock.shared.intProperty
+import ai.tock.shared.property
+
+internal object WorkerOnAwsBatchProvider : WorkerOnDemandProvider {
+    override val workerOnDemandType: WorkerOnDemandType
```

Comment:
is it possible to have an enum that enumerates the different possible types?

Reply:
No types are unknown by the module.
The types are discovered when the application is loaded (via ServiceLoader)